unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs

```diff
@@ -266,9 +282,108 @@ void InitializePerlinNoise()
 
     void OnDestroy()
     {
-        if (nftImageTexture != null)
+        if (nftAsset != null)
+        {
+            nftAsset.Dispose();
+        }
+    }
+
+    static IEnumerator FetchNFTImageAsset(string url, Action<INFTAsset> OnSuccess)
+    {
+        string contentType = null;
+        byte[] bytes = null;
+
+        yield return Utils.FetchAsset(url, UnityWebRequest.Get(url), (request) =>
+        {
+            contentType = request.GetResponseHeader("Content-Type");
+            bytes = request.downloadHandler.data;
+        });
+
+        if (contentType != null && bytes != null)
+        {
+            yield return InstantiateNFTAsset(contentType, bytes, OnSuccess);
+        }
+    }
+
+    static IEnumerator InstantiateNFTAsset(string contentType, byte[] bytes, Action<INFTAsset> OnSuccess)
+    {
+        if (contentType == "image/gif")
+        {
+            var gif = new DCLGif();
+            yield return gif.Load(bytes, () =>
+            {
+                gif.Play();
+                OnSuccess?.Invoke(new NFTGifAsset(gif));
+            });
+        }
+        else
+        {
+            var texture = new Texture2D(1, 1);
+            texture.LoadImage(bytes);
+            OnSuccess?.Invoke(new NFTImageAsset(texture));
+        }
+    }
+
+    interface INFTAsset : IDisposable
+    {
+        Texture2D texture { get; }
+        int width { get; }
+        int height { get; }
+    }
+
+    class NFTImageAsset : INFTAsset
+    {
+        private Texture2D _texture;
+
+        public Texture2D texture => _texture;
+        public int width => _texture.width;
+        public int height => _texture.height;
+
+        public void Dispose()
+        {
+            if (_texture != null)
+            {
+                UnityEngine.Object.Destroy(_texture);
+                _texture = null;
+            }
+        }
+
+        public NFTImageAsset(Texture2D t)
+        {
+            _texture = t;
+        }
+    }
+
+    class NFTGifAsset : INFTAsset
```

Comment:
Why are we using the `NFT` moniker here? This means we can't use this code for a non-NFT gif component? If we can, we should choose a more generic name.